core/types/block.go

```diff
@@ -294,7 +296,14 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		}
 	}
 
-	if withdrawals == nil {
+	if config.IsIsthmus(header.Time) {
+		if withdrawals == nil || len(withdrawals) > 0 {
+			log.Crit("expected non-nil empty withdrawals operation list in Isthmus, but got: %v", body.Withdrawals)
+		}
+		b.header.WithdrawalsHash = header.WithdrawalsHash
+		b.withdrawals = slices.Clone(withdrawals)
```

Comment:
Since we already assert it's non-nil and empty, doing a `make(types.Withdrawals, 0)` may be more readable than a `slices.Clone`.